### PR TITLE
Update to rocksdb dependency to 0.16.0

### PR DESCRIPTION
**Pull request by mina86**
Issue: https://github.com/near/nearcore/issues/3266

### CODE REVIEW

**Comment by mina86** [comment]
I’ve given cursory look at the commits between 0.15 and 0.16 and didn’t notice anything breaking. I’ve run `cargo test` and run the node for a while but don’t really know if that’s sufficient testing for a storage library upgrade.

**Comment by ailisp** [comment]
@mina86 The failing is about cargo-deny. We disallow any crate in Cargo.lock have two different versions. When there's third party dependency cause this (in this case, rocksdb 0.16), add the exception to `deny.toml` and make sure `cargo-deny check bans` pass locally

**Review by bowenwang1996 — APPROVED** [review]

**Review by ailisp — APPROVED** [review]

**Comment by bowenwang1996** [comment]
@olonho @frol @khorolets please review this PR

**Review by khorolets — APPROVED** [review]

**Comment by mina86** [comment]
Haha. The number of reviewers is too damn high. ;) @olonho, could you also approve?

File: core/store/src/db.rs [hunk 1]
```diff
@@ -777,10 +777,10 @@ mod tests {
     impl RocksDB {
         #[cfg(not(feature = "single_thread_rocksdb"))]
         fn compact(&self, col: DBCol) {
-            self.db.compact_range_cf::<&[u8], &[u8]>(
+            self.db.compact_range_cf(
                 unsafe { &*self.cfs[col as usize] },
-                None,
-                None,
+                Option::<&[u8]>::None,
```
Comment (olonho) [inline core/store/src/db.rs:782]: Why this change?

File: deny.toml [hunk 2]
```diff
@@ -129,4 +126,9 @@ skip = [
 
     # hashbrown uses an older version
     { name = "ahash", version = "=0.4.7" },
+
+    # rocksdb (transitively through clang-sys) uses this as a build
+    # dependency which conflicts with 0.6.7 that wasmer-engine-native
```
Comment (olonho) [inline deny.toml:131]: Actually do not think that we need `wasmer-engine-native` at all.

File: core/store/src/db.rs (hunk 1 above)
Comment (mina86) [reply core/store/src/db.rs:782]: The type of the argument changed from `&[u8]` to `Option<AsRef<[u8]>>` and with that the compiler could no longer deduce type.

File: deny.toml (hunk 2 above)
Comment (mina86) [reply deny.toml:131]: It’s used in `runtime/near-vm-runner/src/wasmer1_runner.rs`. It’s also a dependency on `wasmer` 1.0.2.

Comment (olonho) [reply deny.toml:131]: OK, that's probably a mistake, but will fix it.

Comment (olonho) [reply deny.toml:131]: cc @ailisp

**Review by olonho — APPROVED** [review]

File: deny.toml (hunk 2 above)
Comment (ailisp) [reply deny.toml:131]: @olonho we provided configuration option for future switching to native engine here: https://github.com/near/nearcore/blob/f453895831ed423ec41628c034d5aae5e1164896/runtime/near-vm-runner/src/wasmer1_runner.rs#L319

We haven't tried wasmer2 has native engine + singlepass support yet, if it's supported that's still a better option than JIT engine. Note the wasmer 2 renaming: native engine -> DylibEngine; JIT engine -> UniversalEngine

Comment (matklad) [reply deny.toml:131]: Yeah, we should remove support for dylib engine, filed https://github.com/near/nearcore/issues/4520 for that.
